dockstore-common/src/test/java/io/dockstore/common/yaml/DockstoreYamlTest.java

```diff
@@ -453,4 +456,30 @@ public void testAuthorHasNameOrOrcid() throws DockstoreYamlHelper.DockstoreYamlE
     private static String replaceAuthors(String text, String replacement) {
         return text.replaceFirst("(?s)- name: Denis.*?affiliation: UCSC", replacement);
     }
+
+    @Test
+    public void testAuthorEmail() throws DockstoreYamlHelper.DockstoreYamlException {
+        DockstoreYamlHelper.readDockstoreYaml(replaceOrcid(DOCKSTORE12_YAML, "email: test@test.com"), true);
+        try {
+            DockstoreYamlHelper.readDockstoreYaml(replaceOrcid(DOCKSTORE12_YAML, "email: bad"), true);
+            fail("Should not pass property validation because the email address is invalid");
+        } catch (DockstoreYamlHelper.DockstoreYamlException ex) {
+            assertTrue(ex.getMessage().toLowerCase().contains("email"));
+        }
+    }
+
+    private static String replaceOrcid(String text, String replacement) {
+        return text.replaceFirst("orcid: [^ ]*?", replacement);
+    }
+
+    @Test
+    public void testWorkflowSubclass() throws DockstoreYamlHelper.DockstoreYamlException {
+        DockstoreYamlHelper.readDockstoreYaml(DOCKSTORE12_YAML.replace("subclass: wdl", "subclass: WDL"), true);
+        try {
+            DockstoreYamlHelper.readDockstoreYaml(DOCKSTORE12_YAML.replace("subclass: wdl", "subclass: BogusWL"), true);
+            fail("Should not pass property validation because the the workflow subclass is invalid");
+        } catch (DockstoreYamlHelper.DockstoreYamlException ex) {
+            assertTrue(ex.getMessage().toLowerCase().contains("subclass"));
+        }
+    }
```

Comment:
It would be to have a test for the duplicate entry names (or maybe there is and I skimmed over it).

Reply:
There's an integration test that triggers this, I think.  Please let me know if you think we need to add something else.